src/_cert.ts

```diff
@@ -40,10 +44,34 @@ export interface TLSCertOptions
   passphrase?: string;
 }
 
+function pathExists(path: string) {
+  return new Promise((resolve) => {
+    fs.access(path, fs.constants.F_OK)
+      .then(() => resolve(true))
+      .catch(() => resolve(false));
+  });
+}
+
 export async function resolveCertificate(
   options: HTTPSOptions,
 ): Promise<Certificate> {
   let https: Certificate;
+
+  if (typeof options === "object" && options.reuse) {
+    // Reuse previously autogenerated certificates if exists
+    const certExists = await pathExists(certificateDirectory + "cert.pem");
+    const keyExists = await pathExists(certificateDirectory + "cert-key.pem");
+
+    if (certExists && keyExists) {
+      const cert = await fs.readFile(certificateDirectory + "cert.pem");
+      const key = await fs.readFile(certificateDirectory + "cert-key.pem");
+      return {
+        cert: cert.toString("utf8"),
+        key: key.toString("utf8"),
```

Comment:
I think we could perform both keys in a single JSON file. Mainly to simplify fs but also having possibility to for example have multiple generated keys cached in one read/write + having more meta

Reply:
[This](https://github.com/unjs/listhen/pull/136/commits/73dc5be64612715d6ac2149dd106a9dce057dd50) commit introduces these change